src/node_worker.cc

```diff
@@ -368,6 +404,12 @@ void Worker::Run() {
       if (exit_code_ == 0 && !stopped)
         exit_code_ = exit_code;
 
+  active_worker_count--;
+  // Ensure that the warning will be emitted again once the worker count
+  // exceeds the limit again.
+  if (active_worker_count < per_process::cli_options->max_worker_thread_count)
+    active_worker_count_warning = true;
```

Comment:
I’m not sure whether this is really all that helpful, tbh. What benefit would warning a second time give?

Reply:
Going back and forth on this. The additional warning can be used to track cases where a long running process may be pathologically exceed the threshold. But you're right that it is of fairly limited value and likely only useful in very specific edge cases.